cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -334,36 +330,39 @@ Status CallCustomCallback(PyObject* callback, PyObject* elem, PyObject** result)
        << ": handler not registered";
     return Status::NotImplemented(ss.str());
   } else {
-    ScopedRef arglist(Py_BuildValue("(O)", elem));
-    *result = PyObject_CallObject(callback, arglist.get());
+    char arg_format[] = "(O)";
+    *result = PyObject_CallMethod(context, method_name, arg_format, elem);
```

Comment:
https://docs.python.org/2/c-api/object.html says that if you are passing only `PyObject *` args then `PyObject_CallMethodObjArgs` is faster.

Might be worth it since this is a pretty performance critical piece of code.

Reply:
Good point, fixed! We still need to try hard to avoid taking this code path since it will be slow in any case. Maybe in the future we can speed it up with Cython.